lib/impl/dataSources/updateCache.js

```diff
@@ -21,7 +21,11 @@ function updateValidAndInvalidDataSources(dataSourcesToSplit, currentInvalidData
   logger.debug("dataSourcesToSplit", dataSourcesToSplit);
   currentInvalidDataSources = currentInvalidDataSources || [];
   var validInvalid = _.partition(dataSourcesToSplit, function(dataSourceUpdateData) {
-    return !dataSourceUpdateData.error;
+    if (typeof dataSourceUpdateData.error === 'function') {
```

Comment:
Hey @davidkirwan. I have a few questions. How do we know that dataSourceUpdateData.error should always be a function? and if it is not a function, the logic is now returning true and that there is no error. If this logic is true that it is a function, it then returns !dataSourceUpdateData.error which is not invoking the function it self.